cli_tools/gce_vm_image_import/importer/api_inflater.go

```diff
@@ -118,9 +136,17 @@ func (inflater *apiInflater) inflate(loggableBuilder *service.SingleImageImportL
 	return pd, ii, err
 }
 
-// Can't cancel the single disk.insert API call
 func (inflater *apiInflater) cancel(reason string) bool {
-	return false
+	inflater.addTraceLog(fmt.Sprintf("apiInflater.inflate is canceled: %v", reason))
+
+	// Send cancel signal
+	inflater.cancelChan <- reason
+
+	// Wait for cleanup to finish
+	inflater.inflatingSem <- "wait"
+	// Don't block inflation() if cancel() is called earlier
+	<-inflater.inflatingSem
```

Comment:
How does this work, with sending something to a channel then immediately reading from it?

Reply:
in golang, a buffered chan behaves as a semaphore. So it's actually a equivalent of
inflatingSem.Acquire()
inflatingSem.Release()

The purpose is to wait for another sem holder (infalte()) to finish its ongoing work, if any. Then release it.

Reply:
Use a chan as semaphore is pretty usual in golang so far as I know. If it's confusion, I can import another sempahore package and avoid using chan as semaphore to make it clear (https://godoc.org/golang.org/x/sync/semaphore).